### ISSUE TITLE
Build process creates invalid and broken HTML mark-up

### ISSUE THREAD

**Issue by jscholes** [w3c/aria-at#528]
I'm currently working on the test plan for the Date Picker Spin Button (PR https://github.com/w3c/aria-at/pull/411).  Looking at the [original reference file](https://github.com/w3c/aria-at/blob/tests/datepicker-spin-button/tests/datepicker-spin-button/reference/2021-10-15_93728/datepicker-spinbuttons.html) obtained from the APG, here's some example structure for a single spin button and its accompanying prev/next buttons:

```html
<div  class="day spinbutton">
  <button type="button" class="decrease" tabindex="-1" aria-label="previous day">
    <svg ...>...</svg>
  </button>

  <div  class="previous" aria-hidden="true">30</div>

  <div  role="spinbutton" tabindex="0" aria-valuenow="1" aria-valuetext="first" aria-valuemin="1" aria-valuemax="31" aria-label="Day">1</div>

  <div class="next" aria-hidden="true">2</div>

  <button type="button" class="increase" tabindex="-1" aria-label="next day">
    <svg ...>...</svg>
  </button>
</div>
```

In summary, there is a button, a div with `aria-hidden`, the `div` representing the spin button, another hidden `div`, and another button.  All of these elements are siblings, and the mark-up is valid.  I've condensed it a bit, but this is the same structure that can be observed on the [actual APG example page](https://w3c.github.io/aria-practices/examples/spinbutton/datepicker-spinbuttons.html).

In contrast, after running `npm run build`, the mark-up becomes completely mangled.  Here is the same portion of the file, but from the [post-build version](https://github.com/w3c/aria-at/blob/tests/datepicker-spin-button/build/tests/datepicker-spin-button/reference/2021-10-15_93728/datepicker-spinbuttons.html):

```html
<div  class="day spinbutton">
  <button type="button" class="decrease" tabindex="-1" aria-label="previous day">
    <svg ...>...</svg>

    <div class="previous" aria-hidden="true">30</div>

    <div  role="spinbutton" tabindex="0" aria-valuenow="1" aria-valuetext="first" aria-valuemin="1" aria-valuemax="31" aria-label="Day">1</div>

    <div class="next" aria-hidden="true">2</div>

    <button type="button" class="increase" tabindex="-1" aria-label="next day">
      <svg ...>...</svg>
```

The first `<button>` (`class="decrease"`) is missing a closing `</button>` tag.  This means that all elements in the rest of the page are children of that button, including the hidden `div`s and spin buttons.  This already makes the page unusable, because interactive elements are nested inside other ones.

Things only get worse given that the problem occurs for all of the other buttons on the page, and the end result is several unclosed `button`s and `div`s.  I have no idea if this will occur for other types of elements.  This issue makes it impossible to determine the validity of the setup scripts for the test plan, and hence represents a complete blocker to the test writing process.

Here's the [validation result for the post-build version](https://validator.w3.org/nu/?showsource=yes&doc=https%3A%2F%2Fraw.githack.com%2Fw3c%2Faria-at%2Ftests%2Fdatepicker-spin-button%2Fbuild%2Ftests%2Fdatepicker-spin-button%2Freference%2F2021-10-15_93728%2Fdatepicker-spinbuttons.html), showing 32 errors.  The post-build version itself can be viewed at:

https://raw.githack.com/w3c/aria-at/tests/datepicker-spin-button/build/tests/datepicker-spin-button/reference/2021-10-15_93728/datepicker-spinbuttons.html

CC @sinabahram / @s3ththompson

**Comment by jscholes** [comment]
I worked through the test case examples for the test plans currently merged into master.  Here are the ones with post-build errors:

* [checkbox-tri-state](https://w3c.github.io/aria-at/build/tests/checkbox-tri-state/reference/checkbox-tri-state.moveFocusBeforeFirstCheckbox.html): 2 errors, relating to `fieldset` and `div` tags.
* [combobox-select-only](https://w3c.github.io/aria-at/build/tests/combobox-select-only/reference/combobox-select-only.moveFocusBeforeCombobox.html): 2 errors, relating to `section` and `div` tags.
* [menu-button-actions](https://w3c.github.io/aria-at/build/tests/menu-button-actions/reference/2021-1-14_12348/menu-button-actions.setFocusBeforeMenuButton.html): 8 errors, relating to `button`, `ul` and `div` elements, as well as the elements with `role="menu"` and `role="menuitem"`.
* [menu-button-actions-active-descendant](https://w3c.github.io/aria-at/build/tests/menu-button-actions-active-descendant/reference/2021-1-14_171136/menu-button-actions-active-descendant.setFocusBeforeMenuButton.html): 9 errors, similar to the previous plan.
* [menubar-editor](https://w3c.github.io/aria-at/build/tests/menubar-editor/reference/2020-11-23_175528/menubar-editor.focusonfirstlink.html): 5 errors, relating to `textarea`, `div` and other elements.
* [modal-dialog](https://w3c.github.io/aria-at/build/tests/modal-dialog/reference/2020-12-17_13353/dialog.moveFocusToAddDeliveryAddressButton.html): 4 errors, releating to `section` and `div` elements.
* [radiogroup-aria-activedescendant](https://w3c.github.io/aria-at/build/tests/radiogroup-aria-activedescendant/reference/2021-3-17_161413/radio-activedescendant.setFocusBeforeRadioGroup.html): 2 errors, relating to `div` and `ul` elements.

7 out of 16 patterns (43.75%) have 2 or more validation errors.

**Comment by mzgoddard** [comment]
I've posted https://github.com/w3c/aria-at/pull/531 which replaces html parser use for creating templates from the example reference html files with a find and replace based on the closing `head`, `main`, and `body` html tags.

**Comment by jscholes** [comment]
Fixed in #531.
